### PR TITLE
modernized/streamlined AVDL->GO compilation

### PR DESCRIPTION
**Pull request by maxtaco**
- use the new `avdlc` node compiler
- compile straight from AVDL, with the .JSON as an implicit intermediate data format
   - but still compile JSON for the JS compilers
- compile into separate .go files, not a monolith

This depends on:
  - node-avdl2json @2.0.0 or above.
  - node-avdl-compiler @1.0.2 or above

### CODE REVIEW

**Comment by maxtaco** [comment]
@chrisnojima do you want to take a look at this?

File: protocol/Makefile [hunk 1]
```diff
@@ -49,7 +47,8 @@ swift-build-stamp: build-stamp
 	date > $@
 
 clean:
-	rm -rf json/*.json ../go/protocol/keybase_v1.go js/keybase_v1.js swift/*
+	rm -rf json/*.json js/keybase_v1.js swift/* *build-stamp
```
Comment (chrisnojima) [inline protocol/Makefile]: should actually be js/* so we get flow-types.js also (nothing to do with this, but it should be)

**Comment by chrisnojima** [comment]
:+1: would it make sense to put a prelude comment which says its a generated piece of code?

**Comment by maxtaco** [comment]
Yeah that makes a lot of sense. I'll update `node-avdl-compiler` to do so.

**Pull request merged**